### PR TITLE
Add missing 'expand' modifier to list in pcre2test manpage

### PR DESCRIPTION
**Pull request by alexdowad**
The 'expand' modifier is mentioned in later sections of the manpage, but is not included in any of the lists of possible modifiers.

### CODE REVIEW

File: doc/pcre2test.1 [hunk 1]
```diff
@@ -654,6 +654,7 @@ heavily used in the test files.
       convert_glob_separator=c  set glob separator character
       convert_length            set convert buffer length
       debug                     same as info,fullbincode
+      expand                    enable repetition feature in subject
```
Comment (carenas) [inline doc/pcre2test.1:657]: "subject" is the data that gets matched against, while `expand` affects the "pattern" or "needle".

maybe a description in imperative form like the ones used for `convert*` that says: `activate preprocessing of the pattern for expansion`?

**Review by carenas — CHANGES_REQUESTED** [review]
Terminology used is confusing, note that while the syntax used is similar, the `expand` modifier is only needed to instruct `pcre2test` to process and create an "expanded" pattern before calling `pcre2_compile()`.

```
  re> /\[a]{3}/B,expand
------------------------------------------------------------------
        Bra
        aaa
        Ket
        End
------------------------------------------------------------------
```

The expansion of the data doesn't need that modifier:

```
  re> /aaa/B
------------------------------------------------------------------
        Bra
        aaa
        Ket
        End
------------------------------------------------------------------
data> \[a]{3}
 0: aaa

```

File: doc/pcre2test.1 (hunk 1 above)
Comment (alexdowad) [reply doc/pcre2test.1:657]: Dear @carenas, thank you for the great comment! One thing seems confusing: you kindly explained that `expand`  affects only the pattern and not the subject, but the output you posted from `pcre2test` seems to show that expansion is done on *both* the pattern and also the subject. Is that incorrect? If so, why does `/aaa/` match `aaa` in `\[a]{3}`?

Once that is clarified, I'm sure we can find a simple but accurate way to express what `expand` does. Of course, readers will still need to look at the detailed description if they want to understand fully.

Comment (carenas) [reply doc/pcre2test.1:657]: > show that expansion is done on _both_ the pattern and also the subject.

correct, and the syntax is similar, but note that there is no need for a modifier for the subject to be expanded, unlike what happens with the pattern when the modifier is missing:

```
  re> /\[a]{3}/B
------------------------------------------------------------------
        Bra
        [a
        ]{3}
        Ket
        End
------------------------------------------------------------------
```

**Comment by PhilipHazel** [comment]
The bottom line is that expansion is always available in subject lines because there is no conflict with the syntax  - and in fact it was implemented long before expansion in the pattern. For the pattern, there has to be a switch to turn it on because the syntax is the same as pattern syntax, as explained in the doc. So the listing should say something like "enable string expansion in pattern".

**Comment by alexdowad** [comment]
@carenas @PhilipHazel Thanks, this is clear now.

**Comment by alexdowad** [comment]
Updated. Please kindly see if this looks better now.

**Review by carenas — APPROVED** [review]
since the autogenerated `pcre2test.txt` was updated why not also do the same  for `doc/html/pcre2test.html'?

see `PrepareRelease`

**Comment by alexdowad** [comment]
Dear @carenas, thanks for pointing me to `PrepareRelease`. I have updated `pcre2test.html` as advised.

**Review by carenas — APPROVED** [review]

**Pull request merged**